lib/src/main/kotlin/com/swmansion/starknet/account/Account.kt

```diff
@@ -144,15 +145,57 @@ interface Account {
      * @param casmContractDefinition a casm representation of cairo 1/2 compiled contract to be declared
      * @param params additional parameters for the transaction
      * @param forFeeEstimate when set to `true`, it changes the version to `2^128+version` so the signed transaction can only be used for fee estimation
+     * @param hashMethod a method to be used to hash the casmContractDefinition
      * @return signed declare transaction payload
      */
     fun signDeclareV3(
         sierraContractDefinition: Cairo1ContractDefinition,
         casmContractDefinition: CasmContractDefinition,
         params: DeclareParamsV3,
         forFeeEstimate: Boolean,
+        hashMethod: HashMethod,
     ): DeclareTransactionV3
 
+    /**
+     * Sign a version 3 declare transaction.
+     *
+     * Prepare and sign a version 3 declare transaction to be executed on Starknet.
+     *
+     * @param sierraContractDefinition a cairo 1/2 sierra compiled definition of the contract to be declared
+     * @param casmContractDefinition a casm representation of cairo 1/2 compiled contract to be declared
+     * @param params additional parameters for the transaction
+     * @param forFeeEstimate when set to `true`, it changes the version to `2^128+version` so the signed transaction can only be used for fee estimation
+     * @return signed declare transaction payload
+     */
+    fun signDeclareV3(
+        sierraContractDefinition: Cairo1ContractDefinition,
+        casmContractDefinition: CasmContractDefinition,
+        params: DeclareParamsV3,
+        forFeeEstimate: Boolean,
+    ): DeclareTransactionV3 {
+        return signDeclareV3(sierraContractDefinition, casmContractDefinition, params, forFeeEstimate, HashMethod.BLAKE2S)
```

Comment:
We shouldn't have method calls at the interface level, e.g. look at `fun executeV3` - they have different signatures, but actual calls are made in the class which implements the interface.